guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/navigator/CommitNavigatorEntry.java

```diff
@@ -36,87 +37,46 @@
 
-        setupContent( versionRecords );
-    }
+    @UiField
+    Div message;
 
-    public void addContent( final List<VersionRecord> content ) {
-        int base = navigator.getRowCount();
-        for ( int i = 0; i < content.size(); i++ ) {
-            final VersionRecord dataContent = content.get( i );
-            createElement( base + i, dataContent );
-        }
-    }
+    @UiField
+    Span author;
 
-    private void setupContent( final List<VersionRecord> content ) {
-        addContent( content );
+    @UiField
+    Span date;
 
-        container.add( navigator );
-    }
+    @UiField
+    SimplePanel revertButtonContainer;
 
-    private void createElement( final int row,
-                                final VersionRecord dataContent ) {
-        int col = 0;
-
-        final Element messageCol = DOM.createDiv();
-        messageCol.addClassName( NavigatorResources.INSTANCE.css().navigatorMessage() );
-        {
-            { //comment
-                final Element message = DOM.createSpan();
-                message.addClassName( NavigatorResources.INSTANCE.css().message() );
-                message.setInnerText( dataContent.comment() );
-                messageCol.appendChild( message );
-            }
-
-            final Element metadata = DOM.createDiv();
-
-            {//author
-                final Element author = DOM.createSpan();
-                author.addClassName( NavigatorResources.INSTANCE.css().author() );
-                author.setInnerText( dataContent.author() + " - " );
-                metadata.appendChild( author );
-            }
-
-            {//date
-                final Element date = DOM.createSpan();
-                date.addClassName( NavigatorResources.INSTANCE.css().date() );
-                DateTimeFormat fmt = DateTimeFormat.getFormat( "yyyy-MM-dd h:mm a" );
-                date.setInnerText( fmt.format( dataContent.date() ) );
-                metadata.appendChild( date );
-            }
-
-            messageCol.appendChild( metadata );
-        }
+    public CommitNavigatorEntry( final VersionRecord versionRecord,
+                                 final ParameterizedCommand<VersionRecord> onRevertCommand ) {
+        initWidget( uiBinder.createAndBindUi( this ) );
+        initWidget( versionRecord,
+                    onRevertCommand );
+    }
 
-        navigator.setWidget( row, col, new Widget() {{
-            setElement( messageCol );
-        }} );
+    private void initWidget( final VersionRecord versionRecord,
+                             final ParameterizedCommand<VersionRecord> onRevertCommand ) {
+        message.getElement().setInnerSafeHtml( makeSafeHtml( versionRecord.comment() ) );
+        author.getElement().setInnerSafeHtml( makeSafeHtml( versionRecord.author() + " - " ) );
+        date.getElement().setInnerSafeHtml( makeSafeHtml( fmt.format( versionRecord.date() ) ) );
```

Comment:
Way better to use UiBinder here, but would it be possible to use a HTML element or just a Label in the ui.xml for these? 
There might be a reason for doing it this way and I am missing it. Just feels more complex and if I just look at the ui.xml it is not clear what is shown there.